### DISCUSSION TITLE
Linux Kernel Space Driver Example

### DISCUSSION THREAD

**Discussion by kmkorym** [OpenAMP/open-amp#539] [Q&A] [answered]
I want to use openamp to communicate on Linux master and RTOS slave.  
In my situation there are some kernel driver needs to use rpmsg to communicate to RTOS directly ( not from userspace).  
Is there any project use this kind of custom rpmsg kernel driver can take as an example?

**Comment by tnmysh** [comment, marked as answer]
> I want to use openamp to communicate on Linux master and RTOS slave.

Terms "master" with "Host" and "slave" with "Remote" are replaced now.

You can check following rpmsg drivers where Linux is Host:
https://github.com/torvalds/linux/blob/master/samples/rpmsg/rpmsg_client_sample.c
https://github.com/torvalds/linux/blob/master/drivers/tty/rpmsg_tty.c

Corresponding Remote side demo is also available where zephyr RTOS is running on remote processor:
https://github.com/OpenAMP/openamp-system-reference/blob/main/examples/zephyr/rpmsg_multi_services/src/main_remote.c

**Reply by tnmysh** [reply]
Here is documentation of demo: https://openamp.readthedocs.io/en/latest/openamp-system-reference/examples/zephyr/rpmsg_multi_services/README.html

**Reply by kmkorym** [reply]
Thank  you!
